includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -264,7 +264,7 @@ public function get_product_data() {
 
 		$data          = [];
 		$items         = [];
-		$product_price = $this->get_product_price( $product );
+		$product_price = (float) $this->get_product_price( $product );
```

Comment:
I'm not sure how this casting would affect the use of `$product_price` later on. To reduce impact, why not checking whether the price is acceptable below without altering this variable?

Reply:
Hmm that might work, but how to define is price _acceptable_ ?

Reply:
I can't think of a scenario where we'd like to display the buttons when the price is empty, so I'd say we wouldn't support products with no price. We could check that if so. I'm referring to products that have an empty `''` price, not a price of `0`. 

Glad to learn if I'm missing a scenario where we do want to support this, though.